File: test/core/vault_test.rb
require_relative '../test_helper'
require_relative '../../lib/guicrypt_tui/core/vault'

describe GuicryptTui::Vault do
  let(:vault) { GuicryptTui::Vault.new(id: 1, name: 'test', path: '.', mount_point: '/tmp') }

  it 'has correct attributes' do
    expect(vault.id).to eq(1)
    expect(vault.name).to eq('test')
    expect(vault.path).to eq('.')
    expect(vault.mount_point).to eq('/tmp')
  end

  it 'converts to hash' do
    expect(vault.to_h).to eq({ id: 1, name: 'test', path: '.', mount_point: '/tmp' })
  end

  it 'is valid if path exists' do
    expect(vault.valid?).to be true
  end
end

File: test/core/vault_test.cc
#include <gtest/gtest.h>
#include "core/vault.h"

using namespace guicrypt;

TEST(VaultTest, HasCorrectAttributes) {
  Vault v{1, "test", ".", "/tmp"};
  EXPECT_EQ(v.id, 1);
  EXPECT_EQ(v.name, "test");
  EXPECT_EQ(v.path, ".");
  EXPECT_EQ(v.mount_point, "/tmp");
}

TEST(VaultTest, IsValidIfPathExists) {
  Vault v{1, "test", ".", "/tmp"};
  EXPECT_TRUE(v.valid());
}

TEST(VaultTest, IsInvalidIfNameEmpty) {
  Vault v{1, "", ".", "/tmp"};
  EXPECT_FALSE(v.valid());
}
